iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
```

Comment:
An explicit constexpr constructor is not required and can be deleted in your use case.
The following line ```static constexpr invalid_t invalid{};``` will use the compiler generated constexpr ctor